custom/enikshay/reports/views.py

```diff
@@ -0,0 +1,45 @@
+from django.http.response import JsonResponse
+from django.utils.decorators import method_decorator
+from django.views.generic.base import View
+
+from corehq.apps.domain.decorators import login_and_domain_required
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.userreports.reports.filters.choice_providers import ChoiceQueryContext
+
+
+class LocationsView(View):
```

Comment:
Can you inherit from the view that we are currently using? I'm currently making changes to it so that it works with location restrictions. This view will get blocked by the location middleware otherwise.